modules/apps/object/object-service/src/main/java/com/liferay/object/internal/entry/util/ObjectEntryUtil.java

```diff
@@ -107,9 +108,19 @@ public static JSONObject getPayloadJSONObject(
 					return null;
 				}
 
-				return _toDTO(
-					originalBaseModel, dtoConverter, dtoConverterRegistry,
-					Collections.emptyMap(), jsonFactory, modelClass, userId);
+				return ObjectEntryThreadLocal.clearExpandoValues(
+					() -> {
+						Map<String, Object> map = _toDTO(
+							originalBaseModel, dtoConverter,
+							dtoConverterRegistry, Collections.emptyMap(),
+							jsonFactory, modelClass, userId);
+
+						map.put(
+							"customFields",
+							ObjectEntryThreadLocal.getExpandoValues());
```

Comment:
@JhossephG ideally this `customFields` should have the same format as when creating the Account DTO [here](https://github.com/carolmariaabb/liferay-portal/blob/a515bd98bc19bd351339e55bd0e44dfd9d0458fc/modules/apps/headless/headless-admin-user/headless-admin-user-impl/src/main/java/com/liferay/headless/admin/user/internal/dto/v1_0/converter/AccountResourceDTOConverter.java#L147-L152).

Can you try to also use the [CustomFieldsUtil](https://github.com/carolmariaabb/liferay-portal/blob/master/modules/apps/headless/headless-delivery/headless-delivery-api/src/main/java/com/liferay/headless/delivery/dto/v1_0/util/CustomFieldsUtil.java) from `headless-delivery-api` here?